### PR TITLE
add the image saving for 3d

### PR DESCRIPTION
**Pull request by PennyQ**
For #99  With 'image saving' icon on the menu bar and `.png`

I haven't set filters for file format in this PR, one concern is what types should be included in the `filter`, the other concern is that if the user doesn't install the `PIL` or `imageio` package then only `.png` format is supported for saving the output. @astrofrog Any ideas for this?

### CODE REVIEW

**Comment by astrofrog** [comment]
@PennyQ - I think it's fine to support only PNG if PIL or imageio are not installed. Formats that it would be nice to support in addition if possible: JPEG, TIFF, and (likely harder) PDF.

**Comment by PennyQ** [comment]
@astrofrog I added a filter to support `.png` `.jpeg` `.tiff` but the `.pdf` is not supported.

I fount a [image_package](https://github.com/glue-viz/glue/blob/80c9df4fb6306c1a580e63e60a4ef4c54c7d4553/glue/core/data_factories/image.py) that is used in saving images for 2D viewers, maybe we will use it for the future :)

File: glue_vispy_viewers/common/toolbar.py [hunk 1]
```diff
@@ -78,6 +86,23 @@ def __init__(self, vispy_widget=None, parent=None):
...
+        # This indicates that the user cancelled
+        if not outfile:
+            pass
+        img = self._vispy_widget.canvas.render()
+        try:
+            file_filter = str(file_filter).split()[0]
+            io.imsave(outfile, img, format=file_filter)
+        except ImportError:
+            # TODO: give out a window to notify that only .png file format is supported
+            outfile += '.png'
```
Comment (astrofrog) [inline glue_vispy_viewers/common/toolbar.py]: Only add the extension if not already present?

**Comment by astrofrog** [comment]
Thanks @PennyQ!

**Pull request merged**